components/activedirectory/domain_user.go

```diff
@@ -0,0 +1,20 @@
+package activedirectory
+
+// DomainUser represents an Active Directory user
+type DomainUser struct {
+	Username string
+	Password string
+}
+
+// WithDomainUser adds a user in Active Directory.
+// Note: We don't need to be a Domain Controller to create new user in AD but we need
+// the necessary rights to modify the AD.
+func WithDomainUser(username, password string) func(params *Configuration) error {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would move all the `With` to an unique file, they're not that many